### ISSUE TITLE
Changes on CompleteDone not properly undone if new line inserted after completion

### ISSUE THREAD

**Issue by micbou** [vim/vim#3027]
Text changes on the `CompleteDone` event may not be properly undone if new line is inserted after completion. Here's a command that reproduces the issue:
```sh
vim -Nu NONE \
    -c "autocmd CompleteDone * call append(0, ['prepend'])" \
    -c "call feedkeys(\"iline\<ESC>o\<C-X>\<C-L>2\<CR>\<ESC>u\", 't')"
```
The first part of the command creates a `CompleteDone`  autocommand that add the line `prepend` at the start of the buffer. The second part consists of the following steps:
- enter insert mode, write `line`, and leave insert mode. The buffer looks like:
  
  ```
  line
  ```   
- enter insert mode on the next line, start line completion (`i_CTRL-X_CTRL-L`), type 2, insert a newline, and leave insert mode. The buffer now looks like:
  ```
  prepend
  line
  line2
  
  ```
- finally, undo the changes. This results in:
  ```
  line2
  ```
  while the expected buffer is
  ```
  line
  ```

**Comment by micbou** [comment]
In the `CompleteDone` autocommand, it should be `call append(0, ['prepend'])` instead of `call append(1, ['prepend'])` (the line must be added before `line`, not after). Updated the post.

**Comment by brammool** [comment]
Fixed by patch 8.1.0256
